packages/paste-website/src/pages/patterns/filter/index.mdx

```diff
@@ -0,0 +1,698 @@
+              Clear all
+            </Button>
+          ) : null}
+        </Box>
+      </Box>
+      <DataGrid data={filteredTableData} />
+    </>
+  )
+}`}
+/>
+
+### Conditional filters
+
+A conditional filter is a type of filter that becomes available based on the initial filter selections made by the user.
+
+<StoryPreview
+  title="Conditional filter example"
+  storyID="website-filterexamples--conditional-filter-group-example"
+  height="600px"
+  code={`
+export const ConditionalFilterPatternExample = ({data}): React.ReactNode => {
+  const [selectedFilters, setSelectedFilters] = React.useState({});
+  const pillState = useFormPillState();
+  const [filteredTableData, setFilteredTableData] = React.useState(data);
+  return (
+    <>
+      <Heading as="h1" variant="heading50">Filter</Heading>
+      <FormPillGroup {...pillState} aria-label="Filters:" size="large">
+        {filterList.map((pill) => {
+          return (
+            <PopoverContainer key={pill} state={popover}>
+              <PopoverButton
+                variant="reset"
+                size="reset"
+                borderRadius="borderRadiusPill"
+              >
+                <FormPill
+                  {...pillState}
+                  selected={isSelected}
+                  onDismiss={
+                    // Remove the filter from the selected filters
+                  }
+                >
+                  {!isSelected ? <PlusIcon decorative /> : null}
+                  <FilterPillView label={filterMap[pill].label} selectedType={isSelected ? pill : null} selectedValue={value} />
+                </FormPill>
+              </PopoverButton>
+              <Popover aria-label={pill} width="size40">
+                <FilterComponent value={value} onApply={onApply} popover={popover} onRemove={onRemove} /> // Create different filter components
+              </Popover>
+            </PopoverContainer>
+          );
+        })}
+      </FormPillGroup>
+      <Box display="flex" justifyContent="space-between" alignItems="center" columnGap="space30" marginTop="space50">
+        <Box display="flex" columnGap="space30">
+          <DetailText marginTop="space0">
+            <Text as="span" color="colorTextWeak" fontSize="fontSize30">
+              {filteredTableData.length} result{filteredTableData.length !== 1 && "s"}
+            </Text>
+          </DetailText>
+          {filteredTableData.length !== data.length ? (
+            <Button variant="link" onClick={handleClearAll}>
+              Clear all
+            </Button>
+          ) : null}
+        </Box>
+      </Box>
+      <DataGrid data={filteredTableData} />
+    </>
+  )
+}`}
+/>
+
+Use it when filter selections naturally lead to additional criteria, or where certain filter formulas are only necessary for very specific use cases.
+
+<DoDont>
+  <Do title="Do" body="Consider the performance implications of loading conditional filters." />
```

Comment:
question for @roxanagomez : What are the performance implications of loading conditional filters?

Reply:
Like if they're wanting to load a whole new set of filters it might take a long time to render the new components, which can affect performance.